tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/advanced/ScalingTest.java

```diff
@@ -10,111 +10,189 @@
+		this.project = connection.getProject(requiredProject.getProjectName());
+	}
+
+	@Test
+	public void testScaleApplicationViaContextMenuOfService() {
+		Service eapService = project.getService(OpenShiftResources.EAP_SERVICE);
+		assertNotNull(eapService);
+
+		// ensure we scale
+		int pods = PodsAreDeployed.getNumberOfCurrentReplicas(project, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER);
+		if (pods == 1) {
+			pods = 2;
+		} else {
+			pods = 1;
+		}
+
 		eapService.select();
-		scaleUp(++amountOfPods);
-		
+		scaleTo(pods);
+		assertPodAmountDesiredEqualsCurrent(pods, project);
+
 		eapService.select();
-		scaleDown(--amountOfPods);
-		
+		scaleUp();
+		assertPodAmountDesiredEqualsCurrent(++pods, project);
+
 		eapService.select();
-		amountOfPods += 2;
-		scaleTo(amountOfPods);
+		scaleDown();
+		assertPodAmountDesiredEqualsCurrent(--pods, project);
 	}
 
 	@Test
 	public void testScaleApplicationViaContextMenuOfReplicationController() {
-		OpenShiftExplorerView explorer = new OpenShiftExplorerView();
-		explorer.open();
-		
-		int amountOfPods = PodsAreDeployed.getNumberOfDesiredReplicas();
-		new WaitUntil(new PodsAreDeployed(DatastoreOS3.PROJECT1_DISPLAYED_NAME, 
-				replicationControllerName, amountOfPods), TimePeriod.LONG);
-		
-		OpenShiftResource replicationController = explorer.getOpenShift3Connection().
-				getProject().getOpenShiftResources(Resource.DEPLOYMENT).get(0);
+		OpenShiftResource replicationController = 
+				project.getOpenShiftResource(Resource.DEPLOYMENT, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER);
+		assertNotNull(replicationController);
 		
+		// ensure we scale
+		int pods = PodsAreDeployed.getNumberOfCurrentReplicas(project, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER);
+		if (pods == 1) {
+			pods = 2;
+		} else {
+			pods = 1;
+		}
+
 		replicationController.select();
-		scaleUp(++amountOfPods);
-		
+		scaleTo(pods);
+		assertPodAmountDesiredEqualsCurrent(pods, project);
+
 		replicationController.select();
-		scaleDown(--amountOfPods);
-		
+		scaleUp();
+		assertPodAmountDesiredEqualsCurrent(++pods, project);
+
 		replicationController.select();
-		amountOfPods += 2;
-		scaleTo(amountOfPods);
+		scaleDown();
+		assertPodAmountDesiredEqualsCurrent(--pods, project);
 	}
 	
-	private void scaleUp(int newAmountOfPods) {
+	@Test
+	public void scaleTo0ShouldWarn() {
+		int pods = PodsAreDeployed.getNumberOfCurrentReplicas(project, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER);
+		new WaitUntil(
+				new PodsAreDeployed(project, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER, pods),
+				TimePeriod.LONG);
+
+		OpenShiftResource replicationController = 
+				project.getOpenShiftResource(Resource.DEPLOYMENT, OpenShiftResources.EAP_APP_REPLICATION_CONTROLLER);
+
+		replicationController.select();
+		scaleTo(0);
+		Matcher<String> confirmationDialogMatcher = new ShellButtonsMatcher(Arrays.asList("Yes", "No"));
```

Comment:
It's really not necessary to implement such robus matcher and trying to get a shell. Please store shell name in OpenShiftLabel.Shell.STOP_DEPLOYMENT and use here new new DefaultShell(OpenShiftLabel.Shell.STOP_DEPLOYMENT));. Because waiting is default, normal, you don't have to wait for the shell, bcs. shell lookup wait for a period of lenght TimePeriod.NORMAL by default.